xarray/tests/test_dataarray.py

```diff
@@ -1458,6 +1458,10 @@ def test_array_interface(self):
         bar = Variable(['x', 'y'], np.zeros((10, 20)))
         assert_equal(self.dv, np.maximum(self.dv, bar))
 
+    def test_astype_attrs(self):
+        mda = self.mda.astype(bool)
```

Comment:
It would be good to test the datatype converting behavior of `astype()` here, too, at least with a few minimal cases. I don't know if we have test coverage for this already.